src/monstergenerator.cpp

```diff
@@ -454,7 +456,15 @@ void MonsterGenerator::load_monster(JsonObject &jo)
         newmon->dies = get_death_functions(jo, "death_function");
         load_special_defense(newmon, jo, "special_when_hit");
         load_special_attacks(newmon, jo, "special_attacks");
+
         newmon->upgrade_min = jo.get_int("upgrade_min", -1);
+        // recalc upgrade_min if given
+        if (newmon->upgrade_min > 0) {
+            const int season_scale = ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"] / 14;
+            const int season_offset = calendar::start.days();
+            newmon->upgrade_min = (season_offset + (newmon->upgrade_min * season_scale))
+                                   / ACTIVE_WORLD_OPTIONS["MONSTER_UPGRADE_FACTOR"];
```

Comment:
It's float or double. Floating point division by 0 doesn't cause a crash. At least not the division itself - infinity or NaN can lead to crash later, if not handled well.